src/pages/fr/core-concepts/endpoints.md

```diff
@@ -0,0 +1,280 @@
+    })
+  };
+}
+```
+
+## Points de terminaison Serveur(Routes d'API)
+
+Tout ce qui est décrit dans la section des points de terminaison statiques peut également être utilisé en mode SSR : Les fichiers peuvent exporter une fonction `get` qui reçoit un [objet context](/fr/reference/api-reference/#endpoint-context) avec des propriétés similaires à l'objet `Astro` global.
+
+Mais, à la différence du mode `static`, lorsque vous configurez le mode `server`, les points de terminaison seront construits à la demande. Ceci déverrouille de nouvelles fonctionnalités qui ne sont pas disponibles au moment de la compilation, et vous permet de construire des routes d'API qui écoutent les demandes et exécutent le code de manière sécurisée sur le serveur au moment de l'exécution.
+
+:::note
+Soyez sûr d'avoir [activé le SSR](/fr/guides/server-side-rendering/#activation-du-mode-ssr-dans-votre-projet) avant d'essayer ces exemples.
+:::
+
+Les points de terminaison serveur peuvent accéder aux `params` sans avoir à exporter `getStaticPaths`, et ils peuvent retourner un objet [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response), permettant d'affecter des codes de statut et des en-têtes :
+
+```js title="src/pages/[id].json.js"
+import { getProduct } from '../db';
+
+export async function get({ params }) {
+  const id = params.id;
+  const product = await getProduct(id);
+
+  if (!product) {
+    return new Response(null, {
+      status: 404,
+      statusText: 'Not found'
+    });
+  }
+
+  return new Response(JSON.stringify(product), {
+    status: 200,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+```
+
+Ceci répondra à n'importe quelle requête qui correspond à la route dynamique. Par exemple, si nous naviguons vers `/helmet.json`, `params.id` sera affecté à `helmet`. Si `helmet` existe dans la base de données product d'exemple, le point de terminaison utilisera un objet `Response` pour renvoyer du JSON et retourner un [code de statut HTTP 200](https://developer.mozilla.org/en-US/docs/Web/API/Response/status). Sinon, il utilisera un objet `Response` pour renvoyer une `404`.
+
+### Méthodes HTTP
+
+En plus de la fonction `get`, vous pouvez exporter une fonction avec le nom de n'importe quelle [méthode HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods). Lorsqu'une requête arrive, Astro vérifiera la méthode et appellera la fonction correspondante.
+
+Vous pouvez aussi exporter une fonction `all` pour faire correspondre n'importe quelle méthode qui n'a pas de fonction exportée correspondante. S'il y a une demande avec une méthode qui ne correspond pas, elle sera redirigée vers la [page 404](/fr/core-concepts/astro-pages/#page-derreur-404-personnalisée).
+
+:::note
+Etant donné que `delete` est un mot réservé en JavaScript, exportez une fonction `del` pour faire correspondre la méthode delete.
+:::
+
+```ts title="src/pages/methods.json.ts"
+export const get: APIRoute = ({ params, request }) => {
```

Comment:
You may consider translating the `message` property, I don't have a good translation to suggest here. Maybe "C'est un GET!", "Ceci est un GET!"? idk